aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -954,4 +954,46 @@ Tensor dist(const Tensor &self, const Tensor& other, Scalar p){
   return at::norm(self - other, p);
 }
 
+bool cpu_equal(const Tensor& self, const Tensor& other) {
+  if (!at::namedinference::are_names_equal(
+        self.unsafeGetTensorImpl(), other.unsafeGetTensorImpl())) {
+    return false;
+  }
+  at::NoNamesGuard guard;
+  TORCH_CHECK(self.device() == other.device(), "Cannot compare two tensors on "
+              "different devices. Got: ", self.device(), " and ", other.device());
+  TORCH_CHECK(self.dtype() == other.dtype(),
+              "Expected object of scalar type ", self.dtype(), " but got scalar type ",
+              other.dtype(), " for argument 'other'");
+  if (!self.is_same_size(other)) {
+    return false;
+  }
+  bool result = true;
```

Comment:
@xcnick 
Should this be atomic<bool>? I updated my comments around 1hr ago, I think your github page is cached.....